odoo/tools/xml_utils.py

```diff
@@ -14,3 +14,31 @@ def check_with_xsd(tree_or_str, xsd_path):
         #import UserError only here to avoid circular import statements with tools.func being imported in exceptions.py
         from odoo.exceptions import UserError
         raise UserError('\n'.join(str(e) for e in xml_errors.error_log))
+
+def xml_node_chain(first_parent_node, tags_list, str_value=None, return_node=None):
```

Comment:
I wonder about the usefulness of not returning the last node every time... did you face a use case where you wanted a node which wasn't the last (or the first, but you already had it before)?